ingest/sqs/router/usecase/router_usecase.go

```diff
@@ -89,6 +90,77 @@ func (r *routerUseCaseImpl) GetBestSingleRouteQuote(ctx context.Context, tokenIn
 	return router.getBestSingleRouteQuote(tokenIn, routes)
 }
 
+// GetCustomQuote implements mvc.RouterUsecase.
+func (r *routerUseCaseImpl) GetCustomQuote(ctx context.Context, tokenIn sdk.Coin, tokenOutDenom string, poolIDs []uint64) (domain.Quote, error) {
+	// TODO: abstract this
+	router := r.initializeRouter()
+
+	candidateRoutes, err := r.handleRoutes(ctx, router, tokenIn.Denom, tokenOutDenom)
+	if err != nil {
+		return nil, err
+	}
+
+	takerFees, err := r.routerRepository.GetAllTakerFees(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	routes, err := r.poolsUsecase.GetRoutesFromCandidates(ctx, candidateRoutes, takerFees, tokenIn.Denom, tokenOutDenom)
+	if err != nil {
+		return nil, err
+	}
+
+	routeIndex := -1
+
+	for curRouteIndex, route := range routes {
+		routePools := route.GetPools()
+
+		// Skip routes that do not match the pool length.
+		if len(routePools) != len(poolIDs) {
+			continue
+		}
```

Comment:
consider in the future adding length to GetRoutesFromCandidates to prevent having to iterate over them again here

Reply:
I'm note sure what recommendation is exactly. We would have to iterate anyways since we want to make sure each pool's ID matches the id of the given list of pool IDs in the same order

Reply:
Will not be blocking on this, please lmk if any follow-ups are needed